libs/model/src/token/CreateToken.command.ts

```diff
@@ -1,19 +1,76 @@
-import { InvalidInput, type Command } from '@hicommonwealth/core';
+import { type Command } from '@hicommonwealth/core';
 import * as schemas from '@hicommonwealth/schemas';
-import { ChainBase } from '@hicommonwealth/shared';
+import { commonProtocol } from '@hicommonwealth/shared';
+import Web3 from 'web3';
 import { models } from '../database';
 import { AuthContext, isAuthorized } from '../middleware';
 import { mustExist } from '../middleware/guards';
+import { tokenCommunityManagerAbi } from '../services/commonProtocol/abi/TokenCommunityManager';
+import { erc20Abi } from '../services/commonProtocol/abi/erc20';
 
-export const CreateTokenErrors = {
-  TokenNameExists:
-    'The name for this token already exists, please choose another name',
-  InvalidEthereumChainId: 'Ethereum chain ID not provided or unsupported',
-  InvalidAddress: 'Address is invalid',
-  InvalidBase: 'Must provide valid chain base',
-  MissingNodeUrl: 'Missing node url',
-  InvalidNode: 'RPC url returned invalid response. Check your node url',
-};
+export async function createTokenHandler(
+  chainNodeId: number,
+  tokenAddress: string,
+  description?: string,
+  iconUrl?: string,
+) {
+  const chainNode = await models.ChainNode.findOne({
+    where: { id: chainNodeId },
+    attributes: ['eth_chain_id', 'url', 'private_url'],
+  });
+
+  mustExist('Chain Node', chainNode);
+
+  const web3 = new Web3(chainNode.private_url || chainNode.url);
+
+  const contracts =
+    commonProtocol.factoryContracts[
+      chainNode.eth_chain_id as commonProtocol.ValidChains
+    ];
+  const tokenManagerAddress = contracts.tokenCommunityManager;
+  const tokenManagerContract = new web3.eth.Contract(
+    tokenCommunityManagerAbi,
+    tokenManagerAddress,
+  );
+
+  let namespace: string;
+  try {
+    namespace = await tokenManagerContract.methods
+      .namespaceForToken(tokenAddress)
+      .call();
+  } catch (error) {
+    throw Error(`Failed to get namespace for token ${tokenAddress}`);
+  }
+
+  const erc20Contract = new web3.eth.Contract(erc20Abi, tokenAddress);
+
+  let name: string;
+  let symbol: string;
+  let totalSupply: bigint;
+
+  try {
+    name = await erc20Contract.methods.name().call();
+    symbol = await erc20Contract.methods.symbol().call();
+    totalSupply = await erc20Contract.methods.totalSupply().call();
+  } catch (e) {
+    throw Error(
+      `Failed to get erc20 token properties for token ${tokenAddress}`,
+    );
+  }
```

Comment:
The issue with this is that it is just pulls part of the properties of the erc20 token, not all of them. If we move it to the protocol util, it will never be reused so I don't really see a point in it